mne/report.py

```diff
@@ -840,7 +851,11 @@ def __init__(self, info_fname=None, subjects_dir=None,
         # boolean to specify if sections should be ordered in natural
         # order of processing (raw -> events ... -> inverse)
         self._sort_sections = False
-
+        raw_psd = {} if raw_psd is True else raw_psd
+        if raw_psd is not False and not isinstance(raw_psd, dict):
+            raise TypeError('raw_psd must be bool or dict, got %s'
+                            % (type(raw_psd),))
+        self._raw_psd = raw_psd
```

Comment:
following the sklearn way to go I would just do

self. raw_psd = raw_psd

after checking it's a bool or dict and do the conversion to dict in the _render_raw method without adding another attribute to the object.